scripts/cmake/utilities.cmake

```diff
@@ -19,3 +22,437 @@ function(mz_target_handle_warnings MZ_TARGET)
+        ${ARGN})
+
+    set(LOCAL_LINK_LIBRARIES)
+
+    # 1. Qt dependencies handling
+
+    # Get list of required Qt dependencies
+    find_package(Qt6 REQUIRED COMPONENTS ${MZ_GENERATE_LINK_LIBRARIES_QT_DEPENDENCIES})
+    foreach(QT_DEPENDENCY ${MZ_GENERATE_LINK_LIBRARIES_QT_DEPENDENCIES})
+        list(APPEND LOCAL_LINK_LIBRARIES "Qt6::${QT_DEPENDENCY}")
+    endforeach()
+
+    # 2. MZ dependencies handling
+
+    list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_MZ_DEPENDENCIES})
+
+    # 3. Rust dependencies handling
+
+    # Build Rust creates and add to list of linkd targets.
+    foreach(RUST_CRATE_PATH ${MZ_GENERATE_LINK_LIBRARIES_RUST_DEPENDENCIES})
+        # The name of the crate target is expected to be the name of the crate folder
+        get_filename_component(CRATE_NAME ${RUST_CRATE_PATH} NAME)
+
+        include(${CMAKE_SOURCE_DIR}/scripts/cmake/rustlang.cmake)
+        add_rust_library(${CRATE_NAME}
+            PACKAGE_DIR ${RUST_CRATE_PATH}
+            BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}
+            CRATE_NAME ${CRATE_NAME}
+        )
+
+        list(APPEND LOCAL_LINK_LIBRARIES ${CRATE_NAME})
+    endforeach()
+
+    # 4. Platform specific dependencies handling
+
+    if(${MZ_PLATFORM_NAME} STREQUAL "ios")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_IOS_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "android")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_ANDROID_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "macos")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_MACOS_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "linux")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_LINUX_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "windows")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_WINDOWS_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "wasm")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_WASM_DEPENDENCIES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "dummy")
+        list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_DUMMY_DEPENDENCIES})
+    else()
+        message(FATAL_ERROR "MZ_PLATFORM_NAME must be set before creating modules.")
+    endif()
+
+    # 5. Finalize the list of all dependencies
+
+    list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_EXTRA_DEPENDENCIES})
+
+    # 6. Test dependency handling
+
+    set(LOCAL_TEST_LINK_LIBRARIES ${LOCAL_LINK_LIBRARIES})
+
+    # Replace dependencies for test dependencies that start with `replace-`
+    set(REPLACER_DEPENDENCIES ${MZ_GENERATE_LINK_LIBRARIES_TEST_DEPENDENCIES})
+    list(FILTER REPLACER_DEPENDENCIES INCLUDE REGEX "^\\r\\e\\p\\l\\a\\c\\e\\-")
```

Comment:
I don't think alphanumeric chars require escaping in CMake Regex, so this could be simplified to `list(FILTER REPLACER_DEPENDENCIES INCLUDE REGEX "^replace-")`